src/services/AnalyticsService.js

```diff
@@ -0,0 +1,152 @@
+import UserService from "./UserService";
+
+export default {
+  // for tracking only when events happen, not tracking any properties
+  trackNoProperties(name, isFakeUser) {
+    if (isFakeUser) return;
+
+    window.analytics.track(name);
+  },
+
+  /* identifying a user - treats this event differently than a track event 
+  because it allows the ability to look at user specific data, and 
+  automatically attributes all following events with this user. 
+  It should be called when a new user is created, when a user is logged in, 
+  and if the parameters about the user that we are tracking are updated (we are
+  not currently tracking any user specific info that can be changed, but if
+  we were to track their school, a new identify call would have to be called
+   when profile is updated) */
+  identify(userData, isFakeUser) {
+    if (isFakeUser) return;
+
+    const listPassed = [];
+    for (var property in userData) {
+      if (userData[property] && userData[property].hasOwnProperty("passed")) {
+        if (userData[property].passed) {
+          listPassed.push(property);
+        }
+      }
+    }
+    window.analytics.identify(userData._id, {
+      referred: userData.referred ? userData.referred : "No",
+      "list of passed": listPassed ? listPassed : [],
+      "is volunteer": userData.isVolunteer ? "volunteer" : "student"
+    });
+  },
+
+  // tracking the information from the feedback form
+  trackFeedback(feedbackComponent, isFakeUser) {
+    if (isFakeUser) return;
+
+    let aggResponses = [];
+    let volunteerScore = 0;
+
+    // creates an array of an agreggate of all responses and subresponses
+    for (var response in feedbackComponent.userResponse) {
+      if (typeof feedbackComponent.userResponse[response] === "object") {
+        for (var subresponse in feedbackComponent.userResponse[response]) {
+          aggResponses.push(
+            feedbackComponent.userResponse[response][subresponse]
+          );
+        }
+      } else {
+        aggResponses.push(feedbackComponent.userResponse[response]);
+      }
+    }
+    // adds to volunteer score
+    if (feedbackComponent.userType === "student") {
+      volunteerScore = aggResponses.reduce(function(acc, val) {
+        return acc + val;
+      }, 0);
+    }
+    // sends information
+    window.analytics.track("feedback", {
+      "session id": feedbackComponent.sessionId,
+      user: feedbackComponent.userType,
+      "student id": feedbackComponent.studentId,
+      "volunteer id": feedbackComponent.volunteerId,
+      "volunteer score": volunteerScore
+      // can get answers to specific response using aggResponses
+    });
+  },
+
+  // tracks when a help session has ended
+  trackSessionEnded(headerComponent, currentSession, isFakeUser) {
```

Comment:
do we need to send the `headerComponent`?

Reply:
nope! removed it!